CasperInterface.sol

```diff
@@ -1,158 +1,124 @@
-    function get_validatos__dynasty_end(uint128 arg0) constant returns (uint128);
-    function get_validators__addr(uint128 arg0) constant returns (address);
-    function get_validators__withdrawal_address(uint128 arg0) constant returns (address);
-    function get_validators__prev_commit_epoch(uint128 arg0) constant returns (uint128);
-    
-    function get_nextValidatorIndex() constant returns (uint128);
-    function get_dynasty() constant returns (uint128);
-    //function get_next_dynasty_wei_delta() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    //function get_second_next_dynasty_wei_delta() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    
-    function get_dynasty_start_epoch(uint128 arg0) constant returns (uint128);
-    function get_dynasty_in_epoch(uint128 arg0) constant returns (uint128);
-    
-    //function get_consensus_messages__cur_cyn_prepares(uint128 arg0, bytes32 arg1) constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    function get_consensus_messages__prepare_bitmap(uint128 arg0, bytes32 arg1, uint128 arg2) constant returns (uint256); //returns num256
-    //function get_consensus_messages__prev_dyn_prepares(uint128 arg0, bytes32 arg1) constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    function get_consensus_messages__ancestry_hash_justified(uint128 arg0, bytes32 arg1) constant returns (bool);
-    //function get_consensus_messages__cur_dyn_commits(uint128 arg0, bytes32 arg1) constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    //function get_consensus_messages__prev_dyn_commits(uint128 arg0, bytes32 arg1) constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    
-    function get_ancestry_hashes(uint128 arg0) constant returns (bytes32);
-    function get_main_has_justified() constant returns (bool);
-    function get_main_hash_finalized() constant returns (bool);
-    //function get_deposit_scale_factor(uint128 arg0) constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    function get_epoch_lenght() constant returns (uint128);
-    function get_current_epoch() constant returns (uint128);
-    function get_last_finalized_epoch() constant returns (uint128);
-    function get_last_justified_epoch() constant returns (uint128);
-    function get_expected_source_epoch() constant returns (uint128);
-    //function get_reward_factor() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    //function get_base_interest_factor() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    //function get_base_penalty_factor() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-    //function get_current_penalty_factor() constant returns (uintDECIMAL); //DECIMAL = uint128 + 10 decimals
-}
+    event Deposit(
+        address indexed _from,
+        uint128 indexed _validator_index,
```

Comment:
We're currently using `int128` in a number of places where it should probably be a `uint`, but currently this and many other vars are signed.  I'm pushing for a refactor to `uints` but haven't done it yet.